libs/core/include/core/byte_array/const_byte_array.hpp

```diff
@@ -129,7 +129,14 @@ class ConstByteArray
 
   explicit operator std::string() const
   {
-    return {char_pointer(), length_};
+    // Avoiding undefined behaviour (C++ standard defines passing nullptr as
+    // input to `std::basic_string(char const*, ...)` as undefined behaviour):
+    if (pointer() != nullptr)
```

Comment:
I am not sure that this is undefined behaviour because the length would be zero in this case. In either case I think it makes sense to have this as an explicit check

Reply:
C++ standard for [std::basic_string<...>](https://en.cppreference.com/w/cpp/string/basic_string/basic_string) says this for the `basic_string( const CharT* s, size_type count, const Allocator& alloc = Allocator() );`:

Reply:
By mistake I have looked up description of a little bit different method on [std::basic_string<...>](https://en.cppreference.com/w/cpp/string/basic_string/basic_string) - variant 5 (`basic_string( const CharT* s, const Allocator& alloc = Allocator() )`) instead of 4 (`basic_string( const CharT* s, size_type count, const Allocator& alloc = Allocator() );`).

It looks that for variant 4) standard just defines: `"The behaviour is undefined if [s, s + count) is not a valid range."`, where the range should be valid (mathematically speaking) if `s` is  `nullptr`.
However, that shall result in to undefined behaviour if `count > 1`, because constructor would start copying a data starting from `nullptr`, unless constructor does additional check for null-ness of the `s` (what is not required by C++ standard). This should not happen in our case because `ConstByteArray` shall ensure that if `pointer()` method returns `nullptr` then `length_` data member shall have zero value.